### DISCUSSION TITLE
[Messenger] get Method Inconsistency Between AMQP and Other Transports

### DISCUSSION THREAD

**Discussion by m2mtech** [symfony/symfony#51738] [Q&A] [answered]
[`AmqpReceiver->get`](https://github.com/symfony/symfony/blob/bf2d7fac81de2c6ef06b20aa210187ae2a7789cd/src/Symfony/Component/Messenger/Bridge/Amqp/Transport/AmqpReceiver.php#L39-L42) returns a generator, pulling all messages from the queue, while [DoctrineReceiver->get](https://github.com/symfony/symfony/blob/bf2d7fac81de2c6ef06b20aa210187ae2a7789cd/src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/DoctrineReceiver.php#L65) and others return an array with only the first message.

I noticed this using the AMQP transport from Symfony in [Ecotone](https://github.com/ecotoneFramework/), and had to address with a [PR](https://github.com/ecotoneframework/ecotone-dev/pull/207).

Could you please clarify if this behavior is intended? If so, could you provide some insight into the design considerations behind these differences? I'm eager to understand whether this is a feature by design or a potential inconsistency.

Thanks for any clarification!

**Comment by MatTheCat** [comment, marked as answer]
There are no inconsistency here: `ReceiverInterface::get` returns an iterable, and quoting the PHPDoc:

> the intention is that it returns only one, or a "small number" of messages each time.

As long as you take this into account, nothing bad should happen.
